### PR TITLE
Short diff summary

### PR DESCRIPTION
**Pull request by kunalspathak**
POC to create a file that captures short summary of diffs, which excludes the Details and example diffs.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

File: src/coreclr/scripts/superpmi.py [hunk 1]
```diff
@@ -2890,20 +2904,31 @@ def replay_with_throughput_diff(self):
...
                 logging.info("  Summary Markdown file: %s", overall_md_summary_file)
 
+            short_md_summary_file = create_unique_file_name(self.coreclr_args.spmi_location, "tpdiff_short_summary", "md")
+
+            if os.path.isfile(short_md_summary_file):
+                os.remove(short_md_summary_file)
+
+            with open(short_md_summary_file, "w") as write_fh:
+                self.write_tpdiff_markdown_summary(write_fh, tp_diffs, base_jit_build_string_decoded, diff_jit_build_string_decoded, False)
+                logging.info("  Summary Markdown file: %s", short_md_summary_file)
```
Comment (BruceForstall) [inline src/coreclr/scripts/superpmi.py:2926]:
```suggestion
                logging.info("  Short Summary Markdown file: %s", short_md_summary_file)                
```

File: src/coreclr/scripts/superpmi.py [hunk 2]
```diff
@@ -2964,21 +2989,22 @@ def write_pivot_section(row):
             write_pivot_section("Overall")
             write_pivot_section("MinOpts")
             write_pivot_section("FullOpts")
-        else:
+        elif include_details:
```
Comment (BruceForstall) [inline src/coreclr/scripts/superpmi.py:2992]: Doesn't this need to remain an "else"?

Comment (BruceForstall) [reply src/coreclr/scripts/superpmi.py:2992]: I guess you don't want to display anything when there are no diffs, which makes sense. There is no "header" here?

**Review by BruceForstall — APPROVED** [review]
Seems reasonable to me. I'm curious how useful this will be. E.g, I find the "example diffs" the most interesting thing to look at.

File: src/coreclr/scripts/superpmi.py (hunk 2 above)
Comment (kunalspathak) [reply src/coreclr/scripts/superpmi.py:2992]: fixed.

**Comment by kunalspathak** [comment]
> Seems reasonable to me. I'm curious how useful this will be. E.g, I find the "example diffs" the most interesting thing to look at.

We can extend this. One of the reason I am skipping "Example diffs" is because of limit of GH comment. And of course, we will store the overal*.md files, so we can always go and check at different location.

File: src/coreclr/scripts/superpmi.py [hunk 3]
```diff
@@ -2494,96 +2504,100 @@ def write_pivot_section(row):
                                 base_metrics[row]["Diffed code bytes"],
                                 diff_metrics[row]["Diffed code bytes"])))
 
+            write_top_context_section()
```
Comment (jakobbotsch) [inline src/coreclr/scripts/superpmi.py:2507]: Isn't this removing the number of contexts the diffs are based on when there are no diffs? We should keep that. It's important information to know that "no diffs" is based on a reasonable amount of information.

Comment (kunalspathak) [reply src/coreclr/scripts/superpmi.py:2507]: Yes, the context information will be present anyway in the overall*.md file. I am creating the short summary so I can post those contents as GH PR comment. There will be link which will take us to the all the information we capture today.

Comment (jakobbotsch) [reply src/coreclr/scripts/superpmi.py:2507]: Ah, I misread the diff. I thought it was removing the information even for `include_details`, but there's also a call to this function below. LGTM then.

**Review by jakobbotsch — APPROVED** [review]

**Comment by jakobbotsch** [comment]
This seems to have broken the output of the jit-analysis details:

<details>
<summary>Image</summary>
image

**Pull request merged**